Makefile

```diff
@@ -69,49 +70,71 @@ build-push-frontend: build-frontend-static
 	docker buildx build --file Dockerfile --progress plain --push --platform linux/arm64,linux/amd64 \
 	--tag $(REPONAME)/$(FRONTEND_DOCKER_IMAGE):$(DOCKER_TAG) .
 
+# Steps to build static binary of query service
+.PHONY: build-query-service-static
+build-query-service-static:
+	@echo "------------------"
+	@echo "--> Building query-service static binary"
+	@echo "------------------"
+	@if [ $(DEV_BUILD) != "" ]; then \
+		cd $(QUERY_SERVICE_DIRECTORY) && \
+		CGO_ENABLED=1 go build -tags timetzdata -a -o ./bin/query-service-${GOOS}-${GOARCH} \
+    	-ldflags "-linkmode external -extldflags '-static' -s -w ${LD_FLAGS} ${DEV_LD_FLAGS}"; \
+	else \
+		cd $(QUERY_SERVICE_DIRECTORY) && \
+		CGO_ENABLED=1 go build -tags timetzdata -a -o ./bin/query-service-${GOOS}-${GOARCH} \
+		-ldflags "-linkmode external -extldflags '-static' -s -w ${LD_FLAGS}"; \
+	fi
+
+.PHONY: build-query-service-static-amd64
+build-query-service-static-amd64:
+	make GOARCH=amd64 build-query-service-static
+
+.PHONY: build-query-service-static-arm64
+build-query-service-static-arm64:
+	make CC=aarch64-linux-gnu-gcc GOARCH=arm64 build-query-service-static
```

Comment:
This doesn't work anywhere else. The targets should work for all systems; otherwise, developers can't build themselves locally.

Reply:
This is used for building ARM image in AMD machine.
Mostly for releasing multi-arch image from CI or local amd machine.

Reply:
I understood the intent. One can't build the docker image locally because this forces to have `aarch64-linux-gnu-gcc`, which wouldn't exist on many systems and the build target now requires `build-query-service-static-all`. You can create the specific target for release, but we should be able to build a docker image locally with the targets we have been using so far to build local images.

Reply:
I am going to approve this but ideally, individual developers should be able to build local images.

Reply:
yes, will create another PR with updated Makefile either using same Dockerfile or another Dockerfile in which `go build` happens inside container.